gcloud-java-storage/src/main/java/com/google/gcloud/storage/CopyWriter.java

```diff
@@ -120,8 +120,12 @@ public RewriteResponse call() {
         serviceOptions,
         BlobId.fromPb(rewriteResponse.rewriteRequest.source),
         rewriteResponse.rewriteRequest.sourceOptions,
-        BlobInfo.fromPb(rewriteResponse.rewriteRequest.target),
+        BlobId.of(rewriteResponse.rewriteRequest.targetBucket,
+            rewriteResponse.rewriteRequest.targetName),
         rewriteResponse.rewriteRequest.targetOptions)
+        .targetInfo(rewriteResponse.rewriteRequest.targetObject != null
+            ? BlobInfo.fromPb(rewriteResponse.rewriteRequest.targetObject) : null)
+        .result(rewriteResponse.result != null ? BlobInfo.fromPb(rewriteResponse.result) : null)
```

Comment:
Wow, did we miss result from capture? Nice catch. did you update the test for it?